File: patpatbot/DocFileData.py
class DocFileData:
    path: str
    tool: str
    pattern_filename: str
    pattern_description: str
    pattern_description_improved: str

    def __init__(self, tool, path, pattern_filename, pattern_description):
        self.tool = tool
        self.path = path
        self.pattern_filename = pattern_filename
        self.pattern_description = pattern_description

    def to_dict(self, key_prefix=""):
        output = {}
        for attr in dir(self):
            if not callable(attr) and not attr.startswith("__"):
                output[key_prefix + attr] = getattr(self, attr)
        return output

File: src/lib/DocFileData.ts
class DocFileData {
    path: string;
    tool: string;
    patternFilename: string;
    patternDescription: string;

    constructor(tool: string, path: string, patternFilename: string, patternDescription: string) {
        this.tool = tool;
        this.path = path;
        this.patternFilename = patternFilename;
        this.patternDescription = patternDescription;
    }

    toDict(keyPrefix = "init__"): { [key: string]: any } {
        const output: { [key: string]: any } = {};
        for (const key in this) {
            if (this.hasOwnProperty(key)) {
                const snakeCaseKey = key.replace(/([a-z])([A-Z])/g, '$1_$2').toLowerCase();
                output[keyPrefix + snakeCaseKey] = this[key];
            }
        }
        return output;
    }
}

export default DocFileData;
